Source/WebCore/Modules/WebGPU/GPUBuffer.cpp

```diff
@@ -32,12 +32,7 @@
 
 namespace WebCore {
 
-GPUBuffer::~GPUBuffer()
-{
-    m_bufferSize = 0;
-    m_backing->destroy();
```

Comment:
For my understanding, at what point is the `m_backing` reclaimed after the GPU is finished with it?

Reply:
`m_backing` should go out of scope resulting in a ref count of 0, so:
```
RemoteBufferProxy::~RemoteBufferProxy()
{
    auto sendResult = send(Messages::RemoteBuffer::Destruct());
    UNUSED_VARIABLE(sendResult);
}
```
will run, causing the GPU process to remove the underlying buffer from the WebGPUObjectHeap.

This would normally cause the underlying Buffer object to have a ref count of zero, except for when it is retained by an `[Render/Compute]PassEncoder` or `BindGroup`. The expectation is it remains alive in this scenario for the lifetime of the bind group or encoder.